routing/payment_lifecycle.go

```diff
@@ -859,12 +861,39 @@ func (p *shardHandler) handleFailureMessage(rt *route.Route,
 		return err
 	}
 
-	// Apply channel update.
+	var (
+		isAdditionalEdge bool
+		policy           *channeldb.ChannelEdgePolicy
+	)
+
+	// Before we apply the channel update, we need to decide whether the
+	// update is for additional (ephemeral) edge or normal edge stored in
+	// db.
+	if p.paySession != nil {
```

Comment:
can this be nil?

Reply:
Yes. When it's called inside `SendToRoute` then we won't have `paySession` since there's no payment lifecycle.

Reply:
Ah, makes sense! Can you add that to the comment?

Reply:
Of course, added!